python/lsst/ts/observatory/control/auxtel/atcs.py

```diff
@@ -908,9 +1007,16 @@ def flush_offset_events(self):
     async def offset_done(self):
         """Wait for offset events.
         """
-        await self.rem.atmcs.evt_allAxesInPosition.next(
-            flush=False, timeout=self.tel_settle_time
-        )
+        while True:
+            in_position = await self.rem.atmcs.evt_allAxesInPosition.next(
+                flush=False, timeout=self.tel_settle_time
+            )
+
+            if in_position.inPosition:
+                self.log.debug("All axes in position.")
+                return in_position.inPosition
+            else:
+                self.log.debug("Telescope not in position.")
```

Comment:
should this not raise an error?

Reply:
No, if `in_position.inPosition = False` it means the telescope is not in position *yet*, so then you proceed to wait for the event again, until it is `True`. Also, note that, when we send an offset command the mount will first issue an `in_position.inPosition=False` (when the offset start) then `in_position.inPosition=True` (when the offset is complete).